### ISSUE TITLE
`show only new packages` triggers dnfdragora crash

### ISSUE THREAD

**Issue by Germano0** [manatools/dnfdragora#172]
If you go through option->preferences->search and you select show only new packages, dnfdragora will crash
dnfdragora-2.1.0-2.fc32.noarch

**Comment by anaselli** [comment]
here seems to be ok, will try to use fc32 later. Could you try moving the configuration file and check it again?

`mv ~/.config/dnfdragora.yaml ~/.config/dnfdragora.yaml.old`

**Comment by anaselli** [comment]
works here too:
![immagine](https://user-images.githubusercontent.com/1303486/97094709-7c66a680-1657-11eb-99f2-93984954ad53.png)

**Comment by Germano0** [comment]
> here seems to be ok, will try to use fc32 later. Could you try moving the configuration file and check it again?
> 
> `mv ~/.config/dnfdragora.yaml ~/.config/dnfdragora.yaml.old`

Still crashing

**Comment by anaselli** [comment]
Ok failed try, are you able to enable logging and in the case attach it here?

**Comment by Germano0** [comment]
> Ok failed try, are you able to enable logging and in the case attach it here?

No because the crashes do not appear in abrt gui

**Comment by anaselli** [comment]
if you run it from terminal can you see anything written?

And if you can't enable logging by options, can you add it with an external editor? For instance:

```
settings:
  always_yes: false
  do not show groups at startup: false
  interval for checking updates: 180
  log:
    directory: /home/YOURHOME/logs
    enabled: true
    level_debug: true
  metadata:
    last_update: 2020-10-24 21:43
    update_interval: 20
  search:
    match_all: true
    newest_only: false
  upgrades as updates: true
view:
  filter: all
  show: groups

```
Change YOURHOME accordingly.

**Comment by Germano0** [comment]
I have not used the flags you just posted. If the following trace is not enough, tell me and I will use them

```
$ dnfdragora 
Try reading configuration file
From ./dnfdragora.yaml
Skipped exception: <[Errno 2] File o directory non esistente: './dnfdragora.yaml'> 
From /etc/dnfdragora/dnfdragora.yaml
Finally read user settings from /home/user/.config/dnfdragora.yaml
Skipped exception: <[Errno 2] File o directory non esistente: '/home/user/.config/dnfdragora.yaml'> 
Logging disabled
<_M_> [ui] YUILoader.cc:50 loadUI(): DISPLAY: ":0"
<_M_> [ui] YUILoader.cc:51 loadUI(): XDG_CURRENT_DESKTOP: "KDE"
<_M_> [ui] YUILoader.cc:52 loadUI(): YUI_PREFERED_BACKEND: ""
<_M_> [ui] YUILoader.cc:86 loadUI(): User-selected UI-plugin: ""
<_M_> [ui] YPath.cc:49 YPath(): Given filename: libyui-gtk.so.8
<_M_> [ui] YSettings.cc:74 progDir(): progDir: ""
<_M_> [ui] YPath.cc:65 YPath(): Preferring subdir: 
<_M_> [ui] YPath.cc:66 YPath(): Subdir given with filename: 
<_M_> [ui] YPath.cc:67 YPath(): Looking for: libyui-gtk.so.8
<_M_> [ui] YPath.cc:138 lsDir(): Looking in /usr/lib64/yui
<_M_> [ui] YPath.cc:118 YPath(): Found libyui-gtk.so.8 in /usr/lib64/yui
<_M_> [ui] YPath.cc:49 YPath(): Given filename: libyui-ncurses.so.8
<_M_> [ui] YSettings.cc:74 progDir(): progDir: ""
<_M_> [ui] YPath.cc:65 YPath(): Preferring subdir: 
<_M_> [ui] YPath.cc:66 YPath(): Subdir given with filename: 
<_M_> [ui] YPath.cc:67 YPath(): Looking for: libyui-ncurses.so.8
<_M_> [ui] YPath.cc:138 lsDir(): Looking in /usr/lib64/yui
<_M_> [ui] YPath.cc:118 YPath(): Found libyui-ncurses.so.8 in /usr/lib64/yui
<_M_> [ui] YPath.cc:49 YPath(): Given filename: libyui-qt.so.8
<_M_> [ui] YSettings.cc:74 progDir(): progDir: ""
<_M_> [ui] YPath.cc:65 YPath(): Preferring subdir: 
<_M_> [ui] YPath.cc:66 YPath(): Subdir given with filename: 
<_M_> [ui] YPath.cc:67 YPath(): Looking for: libyui-qt.so.8
<_M_> [ui] YPath.cc:138 lsDir(): Looking in /usr/lib64/yui
<_M_> [ui] YPath.cc:118 YPath(): Found libyui-qt.so.8 in /usr/lib64/yui
<_M_> [ui] YUILoader.cc:118 loadUI(): Using UI-plugin: "qt"
<_M_> [ui] YSettings.cc:179 loadedUI(): Set loadedUI to "qt"
<_M_> [ui] YSettings.cc:180 loadedUI(): loadedUI is now locked.
<_M_> [ui] YPath.cc:49 YPath(): Given filename: libyui-qt.so.8
<_M_> [ui] YSettings.cc:74 progDir(): progDir: ""
<_M_> [ui] YPath.cc:65 YPath(): Preferring subdir: 
<_M_> [ui] YPath.cc:66 YPath(): Subdir given with filename: 
<_M_> [ui] YPath.cc:67 YPath(): Looking for: libyui-qt.so.8
<_M_> [ui] YPath.cc:138 lsDir(): Looking in /usr/lib64/yui
<_M_> [ui] YPath.cc:118 YPath(): Found libyui-qt.so.8 in /usr/lib64/yui
<_M_> [ui] YUI.cc:76 YUI(): This is libyui 3.3.3
<_M_> [ui] YUI.cc:77 YUI(): Creating UI without threads
<_M_> [qt-ui] YQUI.cc:100 YQUI(): This is libyui-qt 2.47.1
<_M_> [ui] YUI.cc:230 topmostConstructorHasFinished(): Running without threads
<_M_> [qt-styler] QY2Styler.cc:141 loadStyleSheet(): Couldn't open style sheet "/usr/share/libyui/theme/current/wizard/style.qss"
<_M_> [qt-ui] YQUI.cc:296 processCommandLineArgs(): Qt argument: /usr/bin/python3
<_M_> [qt-ui] YQUI.cc:296 processCommandLineArgs(): Qt argument: /usr/bin/dnfdragora
<_M_> [qt-ui] YQUI.cc:439 calcDefaultSize(): Default size: 2150 x 1173
<ERR> [qt-ui] YQUI.cc:262 initUI(): Forcing /libQtGui.so.5 open failed
<_M_> [qt-ui] YQApplication.cc:133 loadPredefinedQtTranslations(): Selected language: it_IT
<WRN> [qt-ui] YQApplication.cc:137 loadPredefinedQtTranslations(): Qt locale directory not set - no translations for predefined Qt dialogs
<_M_> [qt-ui] YQUI.cc:273 initUI(): YQUI initialized. Thread ID: 0x0x7fc84437d740
<_M_> [ui] YSettings.cc:197 loadedUI(): loadedUI: "qt"
<_M_> [ui] YPath.cc:49 YPath(): Given filename: libyui-mga-qt.so.8
<_M_> [ui] YSettings.cc:74 progDir(): progDir: ""
<_M_> [ui] YPath.cc:65 YPath(): Preferring subdir: 
<_M_> [ui] YPath.cc:66 YPath(): Subdir given with filename: 
<_M_> [ui] YPath.cc:67 YPath(): Looking for: libyui-mga-qt.so.8
<_M_> [ui] YPath.cc:138 lsDir(): Looking in /usr/lib64/yui
<_M_> [ui] YPath.cc:118 YPath(): Found libyui-mga-qt.so.8 in /usr/lib64/yui
<_M_> [ui] YPath.cc:49 YPath(): Given filename: libyui-mga-qt.so.8
<_M_> [ui] YSettings.cc:74 progDir(): progDir: ""
<_M_> [ui] YPath.cc:65 YPath(): Preferring subdir: 
<_M_> [ui] YPath.cc:66 YPath(): Subdir given with filename: 
<_M_> [ui] YPath.cc:67 YPath(): Looking for: libyui-mga-qt.so.8
<_M_> [ui] YPath.cc:138 lsDir(): Looking in /usr/lib64/yui
<_M_> [ui] YPath.cc:118 YPath(): Found libyui-mga-qt.so.8 in /usr/lib64/yui
<_M_> [ew] YExternalWidgets.cc:37 YExternalWidgets(): Creating Libyui External Widgets object
<_M_> [mga-qt-ui] YMGA_QCBTable.cc:74 YMGA_QCBTable():  Slection mode 2
<_M_> [ui] YIconLoader.cc:104 findIcon(): system-search.png not found in /usr/share/YaST2/theme/current/icons/ search path, skipping
<_M_> [ui] YIconLoader.cc:104 findIcon(): system-search.png not found in /usr/share/icons/hicolor/ search path, skipping
<WRN> [qt-ui] YQGenericButton.cc:135 setIcon(): Can't load icon ""
<_M_> [ui] YIconLoader.cc:104 findIcon(): application-exit.png not found in /usr/share/YaST2/theme/current/icons/ search path, skipping
<_M_> [ui] YIconLoader.cc:104 findIcon(): application-exit.png not found in /usr/share/icons/hicolor/ search path, skipping
<WRN> [qt-ui] YQMenuButton.cc:114 rebuildMenuTree(): Can't load icon 
<_M_> [ui] YIconLoader.cc:104 findIcon(): dnfdragora.png not found in /usr/share/YaST2/theme/current/icons/ search path, skipping
<_M_> [ui] YIconLoader.cc:104 findIcon(): dnfdragora.png not found in /usr/share/icons/hicolor/ search path, skipping
<WRN> [qt-ui] YQMenuButton.cc:114 rebuildMenuTree(): Can't load icon 
Get root backend. Locked (False)
Start looking for groups
End found 140 groups
Unexpected error: <class 'KeyError'> 
<_M_> [ui] YUILoader.cc:163 deleteUI(): Shutting down UI
<_M_> [ew] YExternalWidgets.cc:116 ~YExternalWidgetsTerminator(): Shutting down mga External Widgets
<_M_> [ew] YMGAWidgetExtensionFactory.cc:49 ~YMGAWidgetFactory(): Factory removed
```

**Comment by anaselli** [comment]
Does it crash with that option only or with other ones too?
KeyError helps, but not enough. Could you please post your '/home/user/.config/dnfdragora.yaml' file?
I'll try with it

**Comment by Germano0** [comment]
I noticed that after this command
> `mv ~/.config/dnfdragora.yaml ~/.config/dnfdragora.yaml.old`

dnfdragona never created a new `~/.config/dnfdragora.yaml` file. I created a new file with flags suggested in https://github.com/manatools/dnfdragora/issues/172#issuecomment-716129931 and dnfdragora never crashed again. I tried to remove again the file, and I noticed that dnfdragora crashed again with all options under Search preferences. So the crashes seem to be originated by a non existing (or corrupted?)  `~/.config/dnfdragora.yaml` file
